packages/byods/src/DataSourceClient.ts

```diff
@@ -0,0 +1,66 @@
+import {DataSourceRequest, DataSourceResponse, HttpClient, ApiResponse} from './types';
+
+/**
+ * Client for interacting with the /dataSource API.
+ */
+export default class DataSourceClient {
+  private httpClient: HttpClient;
```

Comment:
Here we can use those utility we will create from the BaseClient

Reply:
Created SPARK-563120 to handle this.